src/transformer.jl

```diff
@@ -1,10 +1,19 @@
-function load_model(checkpoint_filename)
+function load_model(checkpoint_filename; mmap=true)
+    local config
     open(checkpoint_filename) do file
         config = read(file, Config)
-        weights = TransformerWeights(config)
-        read!(file, weights)
+        if !mmap
+            weights = TransformerWeights(config)
+            read!(file, weights)
+            return config, weights
+        end
+    end
+    if !mmap
```

Comment:
Is this if statement still necessary, given lines 5-9 above?

Reply:
Yeah, since the return above just exits the `open` closure. It is a bit ugly though to have two conditionals checking this, I agree. But I think your suggestion with the header in `mmap` can resolve this by letting me separate the two paths entirely...

Reply:
Oh right, it's the notorious scope gotcha and JuliaLang/julia/#5334. Carry on